homeassistant/components/assist_satellite/entity.py

```diff
@@ -0,0 +1,151 @@
+"""Assist satellite entity."""
+
+from collections.abc import AsyncIterable
+import time
+from typing import Final
+
+from homeassistant.components import stt
+from homeassistant.components.assist_pipeline import (
+    OPTION_PREFERRED,
+    AudioSettings,
+    PipelineEvent,
+    PipelineEventType,
+    PipelineStage,
+    async_get_pipelines,
+    async_pipeline_from_audio_stream,
+    vad,
+)
+from homeassistant.const import EntityCategory
+from homeassistant.core import Context
+from homeassistant.helpers import entity
+from homeassistant.helpers.entity import EntityDescription
+from homeassistant.util import ulid
+
+from .models import AssistSatelliteState
+
+_CONVERSATION_TIMEOUT_SEC: Final = 5 * 60  # 5 minutes
+
+
+class AssistSatelliteEntity(entity.Entity):
+    """Entity encapsulating the state and functionality of an Assist satellite."""
+
+    entity_description = EntityDescription(
+        key="assist_satellite",
+        translation_key="assist_satellite",
+        entity_category=EntityCategory.CONFIG,
+    )
+    _attr_has_entity_name = True
+    _attr_name = None
+    _attr_should_poll = False
+    _attr_state: AssistSatelliteState | None = AssistSatelliteState.LISTENING_WAKE_WORD
+
+    _conversation_id: str | None = None
+    _conversation_id_time: float | None = None
+
+    _run_has_tts: bool = False
+
+    async def _async_accept_pipeline_from_satellite(
+        self,
+        audio_stream: AsyncIterable[bytes],
+        start_stage: PipelineStage = PipelineStage.STT,
+        end_stage: PipelineStage = PipelineStage.TTS,
+        pipeline_entity_id: str | None = None,
+        vad_sensitivity_entity_id: str | None = None,
+        wake_word_phrase: str | None = None,
+    ) -> None:
+        """Triggers an Assist pipeline in Home Assistant from a satellite."""
+        pipeline_id: str | None = None
+        vad_sensitivity = vad.VadSensitivity.DEFAULT
+
+        if pipeline_entity_id:
+            # Resolve pipeline by name
+            pipeline_entity_state = self.hass.states.get(pipeline_entity_id)
+            if (pipeline_entity_state is not None) and (
+                pipeline_entity_state.state != OPTION_PREFERRED
+            ):
+                for pipeline in async_get_pipelines(self.hass):
+                    if pipeline.name == pipeline_entity_state.state:
+                        pipeline_id = pipeline.id
+                        break
+
+        if vad_sensitivity_entity_id:
+            vad_sensitivity_state = self.hass.states.get(vad_sensitivity_entity_id)
+            if vad_sensitivity_state is not None:
+                vad_sensitivity = vad.VadSensitivity(vad_sensitivity_state.state)
+
+        device_id: str | None = None
+        if self.registry_entry is not None:
```

Comment:
When can this be None?

Reply:
Should I use assert instead for the type checker?

Reply:
Yes

Reply:
@joostlek it can be None when there is no config entry / unique ID. Then there is no device registry entry.